src/cython_code/State.pyx

```diff
@@ -372,18 +372,34 @@ cdef class p_State:
     def insert_row(self, row_data, matching_row_idx, row_idx=-1):
         return self.thisptr.insert_row(row_data, matching_row_idx, row_idx)
     def transition(self, which_transitions=(), n_steps=1,
-                   c=(), r=(), max_iterations=-1, max_time=-1):
+                   c=(), r=(), max_iterations=-1, max_time=-1, progress=None):
+         def _progress(percentage):
+              import sys
+              progress = ' ' * 30
+              fill = int(percentage * len(progress))
+              progress = '[' + '=' * fill + progress[fill:] + ']'
+              print('\r{} {:1.2f}%'.format(progress, 100 * percentage), end="")
+              sys.stdout.flush()
```

Comment:
Setting `progress=False` doesn't help to implement a different kind of progress bar.  This is basic separation of concerns: Crosscat is about empirical inference, not about user interface design.  Keep the user interface design separate.